src/amplitude/worker.py

```diff
@@ -37,9 +37,51 @@ def stop(self):
         self.threads_pool.shutdown()
 
     def flush(self):
+        if not self.storage:
+            return None
+
         events = self.storage.pull_all()
-        if events:
-            return self.threads_pool.submit(self.send, events)
+
+        if not events:
+            return None
+
+        batch_size = self.configuration.flush_queue_size
+        batches = []
```

Comment:
```suggestion
        batches = [events[i:i + batch_size] for i in range(0, len(events), batch_size)]
```
Optional style suggestion (I believe the next loop could also be inlined)